tensorflow/java/src/main/java/org/tensorflow/Graph.java

```diff
@@ -77,6 +80,22 @@ public Operation operation(String name) {
   }
 
   /**
+   * Returns all operations that exist in the Graph.
+   */
+  public List<Operation> getOperations() {
```

Comment:
The underlying C API is safe to concurrent modification, so the Java iterator would be too without any additional machinery. Perhaps you have a concern I'm not catching on to, and if so, please do elaborate.